src/transport/TcpRemotingClient.cpp

```diff
@@ -334,8 +333,9 @@ std::shared_ptr<TcpTransport> TcpRemotingClient::CreateTransport(const string& a
 
     //<!callback;
     TcpTransportReadCallback callback = needResponse ? &TcpRemotingClient::static_messageReceived : nullptr;
-
-    tts = TcpTransport::CreateTransport(this, m_enableSsl, m_sslPropertyFile, callback);
+    if(!tts) {
+      tts = TcpTransport::CreateTransport(this, m_enableSsl, m_sslPropertyFile, callback);
+    }
     TcpConnectStatus connectStatus = tts->connect(addr, 0);  // use non-block
```

Comment:
If and only if create transport, call `connect`. watch change of status if status is wait.

Reply:
and manage failure in the creating branch.

Reply:
fixed